src/webgpu/api/operation/rendering/depth.spec.ts

```diff
@@ -0,0 +1,266 @@
+      colorAttachments: [
+        {
+          attachment: colorAttachmentView,
+          storeOp: 'store',
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+      depthStencilAttachment: {
+        attachment: depthTextureView,
+
+        depthLoadValue,
+        depthStoreOp: 'store',
+        stencilLoadValue: 0,
+        stencilStoreOp: 'store',
+      },
+    });
+    pass.setPipeline(pipeline);
+    pass.draw(3, 2);
+    pass.endPass();
+    encoder.copyTextureToBuffer(
+      { texture: colorAttachment, mipLevel: 0, origin: { x: 0, y: 0, z: 0 } },
+      { buffer: dst, bytesPerRow: 256 },
+      { width: 1, height: 1, depthOrArrayLayers: 1 }
+    );
+    t.device.queue.submit([encoder.finish()]);
+
+    t.expectContents(dst, new Uint8Array(expected));
+  });
+
+g.test('reverse_depth')
+  .desc(
+    `Tests simple rendering with reversed depth buffer, ensures depth test works properly: triangles are in correct order and out of range triangles are clipped.
+    Note that in real use case the depth range remapping is done by the modified projection matrix.
+(see https://developer.nvidia.com/content/depth-precision-visualized).`
+  )
+  .cases(pbool('reversed'))
+  .fn(async t => {
+    const dst = t.device.createBuffer({
+      size: 4,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    const colorAttachment = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+    const colorAttachmentView = colorAttachment.createView();
+
+    const depthBufferFormat = 'depth32float';
+    const depthTexture = t.device.createTexture({
+      size: { width: 1, height: 1 },
+      format: depthBufferFormat,
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.SAMPLED,
+    });
+    const depthTextureView = depthTexture.createView();
+
+    const pipelineDescriptor: GPURenderPipelineDescriptor = {
+      vertex: {
+        module: t.device.createShaderModule({
+          code: `
+            [[builtin(position)]] var<out> Position : vec4<f32>;
+            [[builtin(vertex_index)]] var<in> VertexIndex : u32;
+            [[builtin(instance_index)]] var<in> InstanceIndex : u32;
+
+            [[location(0)]] var<out> color : vec4<f32>;
+
+            [[stage(vertex)]] fn main() -> void {
+              const pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
+                  vec2<f32>(-1.0, -3.0),
+                  vec2<f32>(3.0, 1.0),
+                  vec2<f32>(-1.0, 1.0));
+              // TODO: remove workaround for Tint unary array access broke
+              const zv : array<vec2<f32>, 4> = array<vec2<f32>, 4>(
+                  vec2<f32>(0.2, 0.2),
+                  vec2<f32>(0.3, 0.3),
+                  vec2<f32>(-0.1, -0.1),
+                  vec2<f32>(1.1, 1.1));
+              const z : f32 = zv[InstanceIndex].x;
```

Comment:
nit: Seems like InstanceIndex is only 0 or 1 so some of the entries in `zv` and `colors` are unused?

Reply:
oops. Fixed draw instance count to 4 to utilize those.